scripts/feature_tests/upgrade/combined_tests/1cp_1w_bootDiskImageANDK8sControllers_clusterLevel_upgrade.sh

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash
+
+set -x
+
+IRONIC_IMAGE_TAG=master
+
+METAL3_DEV_ENV_DIR="$(dirname "$(readlink -f "${0}")")/../../../../"
+# shellcheck disable=SC1091
+# shellcheck disable=SC1090
+source "${METAL3_DEV_ENV_DIR}/scripts/feature_tests/upgrade/upgrade_common.sh"
+
+echo '' >~/.ssh/known_hosts
+
+start_logging "${1}"
+# Provision original nodes
+set_number_of_master_node_replicas 1
+set_number_of_worker_node_replicas 1
+
+provision_controlplane_node
+
+controlplane_is_provisioned
+controlplane_has_correct_replicas 1
+
+# apply CNI
+apply_cni
+
+provision_worker_node
+worker_has_correct_replicas 1
+
+# ----------- upgrade controlplane components ---------------
+cleanup_clusterctl_configuration
+
+buildClusterctl
+
+# Install initial version folder structure
+pushd /tmp/cluster-api-clone || exit
+cmd/clusterctl/hack/local-overrides.py
+popd || exit
+
+create_clusterctl_configuration
+
+# Create a new version
+createNextVersionControllers
+
+makeCrdChanges
+
+# show upgrade plan
+clusterctl upgrade plan
+# do upgrade
+clusterctl upgrade plan | grep "upgrade apply" | xargs | xargs clusterctl
+# shellcheck disable=SC2082
+# Verify upgrade
+upgraded_controllers_count=$(kubectl api-resources | grep -Ec "kcp2020|ma2020")
+upgraded_bootstrap_crd_count=$(kubectl get crds \
+  kubeadmconfigs.bootstrap.cluster.x-k8s.io -o json | jq '.spec.names.singular' | wc -l)
+upgraded_capm3_controller_count=$(kubectl api-resources | grep -c m3c2020)
+
+if [ "${upgraded_controllers_count}" -ne 2 ]; then
+  log_error "Failed to upgrade cluster-api and controlplane components"
+  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
+  exit 1
+fi
+if [ "${upgraded_bootstrap_crd_count}" -ne 1 ]; then
+  log_error "Failed to upgrade control-plane-kubeadm components"
+  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
+  exit 1
+fi
+
+if [ "${upgraded_capm3_controller_count}" -ne 1 ]; then
+  log_error "Failed to upgrade infrastructure components"
+  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
+  exit 1
+fi
+
+sleep 30 # Wait for the controllers to be up and running
+
+health_controllers=$(kubectl get pods -A | grep -E "capm3-system|capi-kubeadm|metal3" |
+  grep -vc 'Running')
+if [ "${health_controllers}" -ne 0 ]; then
+  log_error "Some of the upgraded controlplane components are not healthy"
+  log_test_result "1cp_1w_bootDiskImageANDK8sCotrollers_clusterLevel_upgrade.sh" "fail"
+  exit 1
+fi
+
+# ----------- upgrade ironic image ------
+
+# Upgrade container images
+for container in ironic ironic-dnsmasq ironic-httpd mariadb; do
```

Comment:
```suggestion
for image in ironic ironic-dnsmasq ironic-httpd mariadb; do
```

Reply:
hmm, both would be fine in my opinion. I'll keep the container now.